### PR TITLE
SignalR: Clarify groups

### PR DESCRIPTION
**Pull request by wadepickett**
Fixes: #29736

Clarifying:

- Group data is kept in-memory and do not persist.
- You are allowed to add users multiple times to a group, it will not throw an exception.

Also a little bit of cleanup.

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [aspnetcore/signalr/groups.md](https://github.com/dotnet/AspNetCore.Docs/blob/faf6cffea9e0b97634a63aa3027efa0b69194eb1/aspnetcore/signalr/groups.md) | [Manage users and groups in SignalR](https://review.learn.microsoft.com/en-us/aspnet/core/signalr/groups?branch=pr-en-us-32261) |

### CODE REVIEW

File: aspnetcore/signalr/groups.md [hunk 1]
```diff
@@ -32,13 +32,21 @@ Send a message to a specific user by passing the user identifier to the `User` f
...
+Connections are added to or removed from groups via the `AddToGroupAsync` and `RemoveFromGroupAsync` methods:
 
 [!code-csharp[Hub methods](groups/sample/Hubs/ChatHub.cs?range=15-27)]
 
-Group membership isn't preserved when a connection reconnects. The connection needs to rejoin the group when it's re-established. It's not possible to count the members of a group, since this information is not available if the application is scaled to multiple servers.
+It's safe to add a user to a group multiple times, no exception is thrown in the case that the user already exists in the group.
+
+Group membership isn't preserved when a connection reconnects. The connection needs to rejoin the group when it's re-established. It's not possible to count the members of a group, since this information isn't available if the application is scaled to multiple servers. 
+
+Groups are kept in-memory, so they won't persist through a server restart. Consider the Azure SignalR service for scenarios requiring group membership to be persisted. For more information, see [Azure SignalR](https://learn.microsoft.com/azure/azure-signalr/signalr-overview)
```
Comment (tdykstra) [inline aspnetcore/signalr/groups.md:47]:
```suggestion
Groups are kept in memory, so they won't persist through a server restart. Consider the Azure SignalR service for scenarios requiring group membership to be persisted. For more information, see [Azure SignalR](/azure/azure-signalr/signalr-overview)
```

File: aspnetcore/signalr/groups.md [hunk 2]
```diff
@@ -32,13 +32,21 @@ Send a message to a specific user by passing the user identifier to the `User` f
...
 
-To protect access to resources while using groups, use [authentication and authorization](xref:signalr/authn-and-authz) functionality in ASP.NET Core. If a user is added to a group only when the credentials are valid for that group, messages sent to that group will only go to authorized users. However, groups are not a security feature. Authentication claims have features that groups do not, such as expiry and revocation. If a user's permission to access the group is revoked, the app must remove the user from the group explicitly.
+To protect access to resources while using groups, use [authentication and authorization](xref:signalr/authn-and-authz) functionality in ASP.NET Core. If a user is added to a group only when the credentials are valid for that group. This way, messages dispatched to the group are guaranteed to reach only those users who are authorized. However, groups aren't a security feature. Authentication claims have features that groups don't, such as expiry and revocation. If a user's permission to access the group is revoked, the app must remove the user from the group explicitly.
```
Comment (tdykstra) [inline aspnetcore/signalr/groups.md:49]: Something's missing in the sentence fragment "If a user is added to a group only when the credentials are valid for that group."

**Review by tdykstra — APPROVED** [review]
Looks good aside from a couple issues I noted.

File: aspnetcore/signalr/groups.md [hunk 3]
```diff
@@ -32,13 +32,21 @@ Send a message to a specific user by passing the user identifier to the `User` f
...
```
Comment (wadepickett) [inline aspnetcore/signalr/groups.md:49]:
```suggestion
To protect access to resources while using groups, use [authentication and authorization](xref:signalr/authn-and-authz) functionality in ASP.NET Core. If a user is added to a group only when the credentials are valid for that group, messages sent to that group will only go to authorized users. However, groups are not a security feature. Authentication claims have features that groups don't, such as expiry and revocation. If a user's permission to access the group is revoked, the app must remove the user from the group explicitly.
```

**Pull request merged**